src/network/networkd-link.c

```diff
@@ -689,7 +689,9 @@ static Address* link_find_dhcp_server_address(Link *link) {
 static int link_enter_configured(Link *link) {
         assert(link);
         assert(link->network);
-        assert(link->state == LINK_STATE_SETTING_ROUTES);
+
+        if (link->state != LINK_STATE_SETTING_ROUTES)
+                return 1;
```

Comment:
Couple of reasons:
- 0 is the standard value that means "success"
- sometimes we return 0 and 1 to mean, respectively, "success, no action taken" and "success, action taken". This can be useful when the caller needs to distinguish those two cases. For this function, the only caller does not care about the result at all, so there's no point in making this distinction. Also, the convention is reverse (1 is never used for the case of "nothing needs to be done").